app-crypt/pius/pius-3.0.0.ebuild

```diff
@@ -0,0 +1,26 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7,8} )
+
+inherit distutils-r1
+
+DESCRIPTION="A tool for signing and email all UIDs on a set of PGP keys"
+HOMEPAGE="https://www.phildev.net/pius/"
+SRC_URI="https://github.com/jaymzh/pius/releases/download/v${PV}/pius-${PV}.tar.bz2"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+DEPEND=">=app-crypt/gnupg-2.0.0"
+RDEPEND="${DEPEND}
+	dev-lang/perl"
+
+src_test() {
+	"${S}"/run_tests || die "Tests failed"
+}
```

Comment:
You'll want to use `python_test` here so it runs the test phase through on every python implementation installed and enabled. Now it only uses your default one. Check the content of *run_tests* file, it looks like `python3 -m unittest discover -s test/* -v` where you can just replace `python3` with `${EPYTHON}`. However this test looks so simple you can most likely get it to work by calling `distutils_enable_test unittests` from *distutils-r1.eclass* that you use.

Reply:
Unfortunately it looks like `distutils_enable_tests unittests` doesn't provide a way to pass in `-s [arg]` https://github.com/gentoo/gentoo/blob/990a108c6aaae27d968207b6b75da9eb5a9d0cd9/eclass/distutils-r1.eclass#L422-L428, and then `python3 -m unittest discover` finds 0 tests to run:

```
>>> Test phase: app-crypt/pius-3.0.0
 * python3_7: running distutils-r1_run_phase python_test

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
 * python3_8: running distutils-r1_run_phase python_test

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
```

`touch test/__init__.py && touch test/libpius/__init__.py` makes `python -m unittest discover` just work (https://stackoverflow.com/a/29715336/2469311), which might be worth submitting upstream.

With this `python_test`:

```bash
python_test() {
       ${EPYTHON} -m unittest discover -s test/* -v
}
```

tests are run on all enabled python versions:

```
>>> Test phase: app-crypt/pius-3.0.0
 * python3_7: running distutils-r1_run_phase python_test
test_convert_from_v1 (test_state.TestFileConversion) ... ok
test_convert_from_v2 (test_state.TestFileConversion) ... ok
test_read_empty_writes_v3_with_new_data (test_state.TestFileConversion) ... ok
test_read_empty_writes_v3_with_no_new_data (test_state.TestFileConversion) ... ok
test_read_v1_writes_v3 (test_state.TestFileConversion) ... ok
test_read_v2_writes_v3 (test_state.TestFileConversion) ... ok

----------------------------------------------------------------------
Ran 6 tests in 0.013s

OK
 * python3_8: running distutils-r1_run_phase python_test
test_convert_from_v1 (test_state.TestFileConversion) ... ok
test_convert_from_v2 (test_state.TestFileConversion) ... ok
test_read_empty_writes_v3_with_new_data (test_state.TestFileConversion) ... ok
test_read_empty_writes_v3_with_no_new_data (test_state.TestFileConversion) ... ok
test_read_v1_writes_v3 (test_state.TestFileConversion) ... ok
test_read_v2_writes_v3 (test_state.TestFileConversion) ... ok

----------------------------------------------------------------------
Ran 6 tests in 0.019s

OK
```

Also note that the unit tests themselves all pass on python 3.6, which should probably also be reported upstream.